data_structures/stacks/postfix_evaluation.py

```diff
@@ -17,52 +24,211 @@
+
+    Parameters
+    ----------
+    data : str
+        The data which needs to be converted to the appropriate number
+
+    Returns
+    -------
+    bool,  int or float
+        Returns a tuple of (a, b) where 'a' is True if data is indeed a number (integer
+        or numeric) and 'b' is either an integer of a floating point number.
+        If 'a' is False, then b is 'data'
+    """
+    try:
+        return int(data)
+    except ValueError:
+        try:
+            return float(data)
+        except ValueError:
+            if is_operator(data):
+                return data
+            msg = f"{data} is neither a number nor a valid operator"
+            raise ValueError(msg)
+
+
+def is_operator(data: str | int | float) -> bool:
+    """
+    Checks whether a given input is one of the valid operators or not.
+    Valid operators being '-', '+', '*', '^' and '/'.
+
+    Parameters
+    ----------
+    data : str
+        The value that needs to be checked for operator
+
+    Returns
+    -------
+    bool
+        True if data is an operator else False.
+    """
+    return data in BINARY_OP_SYMBOLS
+
+
+def evaluate(post_fix: list[str], verbose: bool = False) -> int | float | str | None:
+    """
+    Function that evaluates postfix expression using a stack.
+    >>> evaluate(["2", "1", "+", "3", "*"])
+    9
+    >>> evaluate(["4", "13", "5", "/", "+"])
+    6.6
+    >>> evaluate(["2", "-", "3", "+"])
+    1
+    >>> evaluate(["-4", "5", "*", "6", "-"])
+    -26
+    >>> evaluate([])
+    0
+
+    Parameters
+    ----------
+    post_fix : list
+        The postfix expression tokenized into operators and operands and stored as a
+        python list
 
+    verbose : bool
+        Display stack contents while evaluating the expression if verbose is True
 
-def solve(post_fix):
+    Returns
+    -------
+    int
+        The evaluated value
+    """
+    x: str | int | float
     stack = []
-    div = lambda x, y: int(x / y)  # noqa: E731 integer division operation
+    valid_expression = []
     opr = {
-        "^": op.pow,
-        "*": op.mul,
-        "/": div,
-        "+": op.add,
-        "-": op.sub,
+        "^": lambda p, q: p**q,
+        "*": lambda p, q: p * q,
+        "/": lambda p, q: p / q,
+        "+": lambda p, q: p + q,
+        "-": lambda p, q: p - q,
     }  # operators & their respective operation
-
-    # print table header
-    print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
-    print("-" * (30 + len(post_fix)))
-
+    if len(post_fix) == 0:
+        return 0
+    # Checking the list to find out whether the postfix expression is valid
     for x in post_fix:
-        if x.isdigit():  # if x in digit
+        valid_expression.append(get_number(x))
+    if verbose:
+        # print table header
+        print("Symbol".center(8), "Action".center(12), "Stack", sep=" | ")
+        print("-" * (30 + len(post_fix)))
+    for x in valid_expression:
+        if not is_operator(x):
```

Comment:
We do not need `is_operator()` if we use:
```suggestion
        if x not in opr:  # if x is not an operator
```